src/hyperion/parameters/beamline_parameters.py

```diff
@@ -30,22 +32,55 @@ def from_lines(cls, config_lines: list[str]):
             if len(param) == 2
         ]
         for i, (param, value) in enumerate(config_pairs):
-            if value == "Yes":
-                config_pairs[i] = (config_pairs[i][0], True)
-            elif value == "No":
-                config_pairs[i] = (config_pairs[i][0], False)
-            elif value in BEAMLINE_PARAMETER_KEYWORDS:
-                pass
-            else:
-                config_pairs[i] = (config_pairs[i][0], float(config_pairs[i][1]))
+            try:
+                # BEAMLINE_PARAMETER_KEYWORDS effectively raw string but whitespace removed
+                if value not in BEAMLINE_PARAMETER_KEYWORDS:
+                    config_pairs[i] = (
+                        config_pairs[i][0],
+                        cls.parse_value(config_pairs[i][1]),
```

Comment:
NIT: maybe slightly easier to read if this is cls.parse_value(value)

Reply:
Do the tests pass for you locally @olliesilvester? for some reason everything is broken for me, only on this branch - but I see they have passed in CI.

Reply:
No, they're not passing for me either

Reply:
Ok thanks, I'll try to figure out what's going on - it's weird, seems unrelated to the code that was touched here but it really is specifically on this branch

Reply:
Ah, it was because the parameters hadn't been updated for the dodal fix which stopped noninteger dwell times - the test data has been fixed in main, but the error reporting is poor so I've made an issue for that https://github.com/DiamondLightSource/hyperion/issues/972